src/arr/trove/statistics.arr

```diff
@@ -67,6 +68,80 @@ fun median(l :: L.List):
   end
 end
 
+fun num-counts(lst :: L.List<Number>) -> sd.StringDict<Number>:
+  doc: "Returns dictionary mapping appearances of each list element"
+  cases (L.List) lst:
+    | empty => [sd.string-dict: ]
+    | link(f, r) => 
+      rest = num-counts(r)
+      key = num-to-string(f)
+      val = rest.get(key)
+
+      cases (O.Option) val:
+        | none => rest.set(key, 1)
+        | some(v) => rest.set(key, v + 1)
+      end
+  end 
+end
+
+fun force-unwrap-string(s :: String) -> Number:
+  cases (O.Option) string-to-number(s):
+    | none => raise("Mode does not appear in list") # Mode counts construction prevents this 
+    | some(v) => v
+  end
+end
+
+fun modes(lst :: L.List<Number>) -> L.List<Number>:
```

Comment:
@jswrenn `Set` can't contain rough-nums, so even though all of the elements will be distinct, I think it needs to be `List`